src/types/index.d.ts

```diff
@@ -0,0 +1,12 @@
+// TODO proper type for Vue Router routes when that package exports types (missing in rc.2)
+
+type ViteSSROptions = { routes: any[] }
+
+type Hook = (params: {
+  app: any;
+  router: any;
+  request?: Request;
+  baseUrl?: string;
+  isClient: boolean;
+  initialState?: Record<string, any>;
+}) => Promise<void>;
```

Comment:
i saw that the hook was different for different files. Should those be entirely separate types?

Reply:
Yeah maybe better to have different types, one for entry-server and one for entry-client 🤔 Just to make the types more specific.

Reply:
done @frandiox